File: app/src/main/java/com/mailerdaemon/app/Clubs/ClubAdapter.java
package com.mailerdaemon.app.Clubs;

import android.content.Context;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import androidx.swiperefreshlayout.widget.CircularProgressDrawable;

import com.facebook.drawee.view.SimpleDraweeView;
import com.mailerdaemon.app.R;

import java.util.ArrayList;
import java.util.List;

import Utils.DialogOptions;

public class ClubAdapter extends RecyclerView.Adapter<ClubAdapter.Holder> {
    private List<ClubIconModel> iconModel=new ArrayList<>();
    private CircularProgressDrawable drawable;
    private DialogOptions options;

    ClubAdapter(Context context, DialogOptions options){
    this.options=options;
    }

    @NonNull
    @Override
    public Holder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_club,parent,false);
        return new Holder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull Holder holder, int position) {
        ClubIconModel model= iconModel.get(position);
        //holder.icon.getHierarchy().setProgressBarImage(drawable);
        if (model.getUrl()==null){
            holder.icon.setImageResource(R.drawable.ic_account_circle_black_24dp);
        }else{//holder.icon.setBackground(null);
    holder.icon.setImageURI(Uri.parse(model.getUrl()));}
        holder.icon.setOnClickListener(v-> options.showDialog(model.getTag()));
    }

    @Override
    public int getItemCount() {
        return iconModel.size();
    }

    public void setData(List<ClubIconModel> iconModel) {
        this.iconModel=iconModel;
    }

    public static class Holder extends RecyclerView.ViewHolder {
        SimpleDraweeView icon;

        public Holder(@NonNull View itemView) {
            super(itemView);
            icon=itemView.findViewById(R.id.club_icon);
        }
    }
}

File: app/src/main/java/com/mailerdaemon/app/clubs/ClubAdapter.kt
package com.mailerdaemon.app.clubs

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.facebook.drawee.view.SimpleDraweeView
import com.mailerdaemon.app.R
import com.mailerdaemon.app.utils.DialogOptions
import java.util.*

class ClubAdapter internal constructor(private val options: DialogOptions) :
    RecyclerView.Adapter<ClubAdapter.Holder>() {
    private var iconModel: List<ClubIconModel> = ArrayList()
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Holder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_club, parent, false)
        return Holder(view)
    }

    override fun onBindViewHolder(holder: Holder, position: Int) {
        val (url, tag) = iconModel[position]
        if (url == null) {
            holder.icon.setActualImageResource(R.drawable.ic_account_circle_black_24dp)
        } else {
            var id = 0
            when (tag) {
                1 -> id = R.drawable.md_logo
                2 -> id = R.drawable.litc_logo
                3 -> id = R.drawable.mechismu_logo
                4 -> id = R.drawable.cyberlabs_logo
                5 -> id = R.drawable.chayanika_logo
                6 -> id = R.drawable.fintech_logo
                7 -> id = R.drawable.arka_logo
                8 -> id = R.drawable.quiz_logo
                9 -> id = R.drawable.udaan_logo
                10 -> id = R.drawable.toastmaster_logo
                11 -> id = R.drawable.ffi_logo
                12 -> id = R.drawable.wtc_logo
                13 -> id = R.drawable.kartavya_logo
                14 -> id = R.drawable.rythm_logo
                15 -> id = R.drawable.litm_logo
                16 -> id = R.drawable.lci_logo
                17 -> id = R.drawable.ff_logo
                18 -> id = R.drawable.roboism_logo
                19 -> id = R.drawable.manthan_logo
                20 -> id = R.drawable.adc_logo
                21 -> id = R.drawable.art_logo
                22 -> id = R.drawable.livetalksism_logo
                23 -> id = R.drawable.bookclubiitism_logo
                24 -> id = R.drawable.comclub_logo
                25 -> id = R.drawable.electronicsclub_logo
                26 -> id = R.drawable.smfclub_logo
                27 -> id = R.drawable.animationandgamedesignclub_logo
                28 -> id = R.drawable.kalpitam_logo
            }
            holder.icon.setActualImageResource(id)
        }
        holder.icon.setOnClickListener { options.showDialog(Objects.requireNonNull(tag).toString()) }
    }

    override fun getItemCount(): Int {
        return iconModel.size
    }

    fun setData(iconModel: List<ClubIconModel>) {
        this.iconModel = iconModel
    }

    inner class Holder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var icon: SimpleDraweeView = itemView.findViewById(R.id.club_icon)
    }
}
